torch/_dynamo/variables/builder.py

```diff
@@ -942,6 +942,12 @@ def _clone_input(value):
     elif isinstance(example_value, (torch.SymInt, torch.SymFloat)):
         proxy.node.meta["example_value"] = example_value
         return DynamicShapeVariable(proxy, example_value, **options)
+    elif example_value is NotImplemented:
```

Comment:
Riding NotImplemented in example_value is weird, to me - where is it coming from?

Reply:
The naming of `example_value` was a bit weird to me, but it comes from:
```
example_value = get_fake_value(proxy.node, tx)
```

so it's the result of running the node computation using fake tensors. `get_fake_value()` will return `NotImplemented` for the cases we're interested in.